app/models/repository/Merger.scala

```diff
@@ -425,12 +425,21 @@ object Merger {
               balancingCharge = Some(
                 BalancingCharge(
                   balancingChargeYesNo = extractedMaybe
-                    .map(_.raRBalancingChargeYesNo)
+                    .map(_.balancingCharge.exists(_.balancingChargeYesNo == true))
```

Comment:
I think we need to properly handle balancingChargeYesNo.

```suggestion
               .flatMap(_.balancingCharge.map(_.balancingChargeYesNo))
```